scripts/spack/packages/mfem/package.py

```diff
@@ -525,11 +525,11 @@ def find_optional_library(name, prefix):
         if '+netcdf' in spec:
             lib_flags = ld_flags_from_dirs([spec['netcdf-c'].prefix.lib],
                                            ['netcdf'])
-            hdf5 = spec['hdf5:hl']
-            if hdf5.satisfies('~shared'):
-                hdf5_libs = hdf5.libs
-                hdf5_libs += LibraryList(find_system_libraries('libdl'))
-                lib_flags += " " + ld_flags_from_library_list(hdf5_libs)
+#            hdf5 = spec['hdf5:hl']
+#            if hdf5.satisfies('~shared'):
+#                hdf5_libs = hdf5.libs
+#                hdf5_libs += LibraryList(find_system_libraries('libdl'))
+#                lib_flags += " " + ld_flags_from_library_list(hdf5_libs)
```

Comment:
Netcdf/hdf5 failed to compile on WSL 2. Somehow it failed to find the C compiler 20 packages down the concretizer. 

It would be nice to have a method of better overriding netcdf when things just aren't working. I should probably move this packaging stuff into another PR.